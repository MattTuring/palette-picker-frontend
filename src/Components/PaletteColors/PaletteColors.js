import React from 'react';
import lock from '../../lock.png';
import unlocked from '../../open-lock.png';

const PaletteColors = ({colors, lockeds, setLockeds}) => {
  const {color1, color2, color3, color4, color5} = colors;
  const {locked1, locked2, locked3, locked4, locked5} = lockeds;
  const {setLocked1, setLocked2, setLocked3, setLocked4, setLocked5} = setLockeds;
  
  return (
    <section className="color-picker">
      <div style={{backgroundColor: color1}} className="colors color1">{color1} <img className='lock' onClick={() => setLocked1(!locked1)} src={locked1 ? lock : unlocked} alt='lock'/></div>
      <div style={{backgroundColor: color2}} className="colors color2">{color2} <img className='lock' onClick={() => setLocked2(!locked2)} src={locked2 ? lock : unlocked} alt='lock'/></div>
      <div style={{backgroundColor: color3}} className="colors color3">{color3} <img className='lock' onClick={() => setLocked3(!locked3)} src={locked3 ? lock : unlocked} alt='lock'/></div>
      <div style={{backgroundColor: color4}} className="colors color4">{color4} <img className='lock' onClick={() => setLocked4(!locked4)} src={locked4 ? lock : unlocked} alt='lock'/></div>
      <div style={{backgroundColor: color5}} className="colors color5">{color5} <img className='lock' onClick={() => setLocked5(!locked5)} src={locked5 ? lock : unlocked} alt='lock'/></div>
    </section>
  )
};

export default PaletteColors;
