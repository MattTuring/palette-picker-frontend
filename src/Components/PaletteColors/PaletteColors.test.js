import React from 'react';
import { shallow } from 'enzyme';
import PaletteColors from './PaletteColors';

describe('PaletteColors', () => {
  const setLocked1 = jest.fn();
  const setLocked2 = jest.fn();
  const setLocked3 = jest.fn();
  const setLocked4 = jest.fn();
  const setLocked5 = jest.fn();

  const mockProps = {
    colors: {
      color1: '#000000',
      color2: '#ffffff',
      color3: '#000000',
      color4: '#ffffff',
      color5: '#000000'
    },
    lockeds: {
      locked1: false,
      locked2: true,
      locked3: false,
      locked4: true,
      locked5: false
    },
    setLockeds: {
      setLocked1,
      setLocked2,
      setLocked3,
      setLocked4,
      setLocked5
    }
  };
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <PaletteColors {...mockProps} />);
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call setLocked1 after click on icon for the first color with oposite value of locked', () => {
    wrapper.find('.color1 img').simulate('click');
    expect(setLocked1).toHaveBeenCalledWith(true);
  });

  it('should call setLocked2 after click on icon for the second color with oposite value of locked', () => {
    wrapper.find('.color2 img').simulate('click');
    expect(setLocked2).toHaveBeenCalledWith(false);
  });

  it('should call setLocked3 after click on icon for  third color with oposite value of locked', () => {
    wrapper.find('.color3 img').simulate('click');
    expect(setLocked3).toHaveBeenCalledWith(true);
  });

  it('should call setLocked4 after click on icon for the fourth color with oposite value of locked', () => {
    wrapper.find('.color4 img').simulate('click');
    expect(setLocked4).toHaveBeenCalledWith(false);
  });

  it('should call setLocked5 after click on icon for the fourth color with oposite value of locked', () => {
    wrapper.find('.color5 img').simulate('click');
    expect(setLocked5).toHaveBeenCalledWith(true);
  });
});
