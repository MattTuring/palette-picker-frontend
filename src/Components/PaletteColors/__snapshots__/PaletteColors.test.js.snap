// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteColors should match the snapshot with all data passed in correctly 1`] = `
<section
  className="color-picker"
>
  <div
    className="colors color1"
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    #000000
     
    <img
      alt="lock"
      className="lock"
      onClick={[Function]}
      src="open-lock.png"
    />
  </div>
  <div
    className="colors color2"
    style={
      Object {
        "backgroundColor": "#ffffff",
      }
    }
  >
    #ffffff
     
    <img
      alt="lock"
      className="lock"
      onClick={[Function]}
      src="lock.png"
    />
  </div>
  <div
    className="colors color3"
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    #000000
     
    <img
      alt="lock"
      className="lock"
      onClick={[Function]}
      src="open-lock.png"
    />
  </div>
  <div
    className="colors color4"
    style={
      Object {
        "backgroundColor": "#ffffff",
      }
    }
  >
    #ffffff
     
    <img
      alt="lock"
      className="lock"
      onClick={[Function]}
      src="lock.png"
    />
  </div>
  <div
    className="colors color5"
    style={
      Object {
        "backgroundColor": "#000000",
      }
    }
  >
    #000000
     
    <img
      alt="lock"
      className="lock"
      onClick={[Function]}
      src="open-lock.png"
    />
  </div>
</section>
`;
